File: angular/src/app/models/test-result.ts
export class TestResult {
    id: number;
    authority: string;
    status: number;
    testDate: Date;
    minResponseTime: number;
    maxResponseTime: number;
    meanResponseTime: number;

    constructor(obj: any) {
        const { id, authority, status, testDate, minResponseTime, maxResponseTime, meanResponseTime } = obj;
        const { Id, Authority, Status, TestDate, MinResponseTime, MaxResponseTime, MeanResponseTime } = obj;
        this.id = this.isDefined(id) ? id : Id;
        this.authority = authority || Authority;
        this.status = this.isDefined(status) ? status : Status;
        this.testDate = new Date(testDate || TestDate);
        this.minResponseTime = this.isDefined(minResponseTime) ? minResponseTime : MinResponseTime;
        this.maxResponseTime = this.isDefined(maxResponseTime) ? maxResponseTime : MaxResponseTime;
        this.meanResponseTime = this.isDefined(meanResponseTime) ? meanResponseTime : MeanResponseTime;
    }

    private isDefined(value: any) {
        if (value !== undefined && value !== null) {
            return true;
        }

        return false;
    }
}

File: Core/Entities/TestResult.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Core.Entities
{
    public class TestResult
    {
        public int Id { get; set; }

        [Required]
        public string Authority { get; set; }
        public int Status { get; set; }
        public DateTime TestDate { get; set; }

        public long MinResponseTime { get; set; }
        public long MaxResponseTime { get; set; }
        public double MeanResponseTime { get; set; }

        public IList<TestResultDetail> TestResultDetails { get; set; }
    }
}
